clang-tools-extra/clang-tidy/bugprone/MethodHidingCheck.cpp

```diff
@@ -0,0 +1,141 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "MethodHidingCheck.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+#include <stack>
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::bugprone {
+
+namespace {
+
+bool sameBasicType(ParmVarDecl const *Lhs, ParmVarDecl const *Rhs) {
+  if (Lhs && Rhs) {
+    return Lhs->getType()
+               .getCanonicalType()
+               .getNonReferenceType()
+               .getUnqualifiedType() == Rhs->getType()
+                                            .getCanonicalType()
+                                            .getNonReferenceType()
+                                            .getUnqualifiedType();
+  }
+  return false;
+}
+
+bool namesCollide(CXXMethodDecl const &Lhs, CXXMethodDecl const &Rhs) {
+  if (Lhs.getNameAsString() != Rhs.getNameAsString()) {
+    return false;
+  }
```

Comment:
```suggestion
  if (Lhs.getNameAsString() != Rhs.getNameAsString())
    return false;
```
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements
Same everywhere

Reply:
There is an option in `clang-format` for it https://clang.llvm.org/docs/ClangFormatStyleOptions.html#removebracesllvm but It's not widely used due to instability I think.

> Would a backwards readability-braces-around-statements check be accepted upstream, with or without fixits?

Pretty much yes, I was thinking of implementing it when I have more time. If you want to work on it - it would be great.

> Reading around, there's a suggestion that removing brackets automatically is considered bad practice.

Is it somewhere in LLVM docs or generally in the web? Could you share a link?

Reply:
I've got one or two more things to upstream before I'd have time to look at implementing this (watch this space...)

I don't have a good reference for this - it's something I found buried on stackoverflow or somewhere, and now I can't find it again.  If you're not aware of it, then it's probably just someone talking nonsense.

Reply:
Anyway, I fixed this manually in this PR, so I'll mark this conversation as resolved